packages/gatsby/src/utils/webpack.config.js

```diff
@@ -86,6 +86,12 @@ module.exports = async (
     envObject.BUILD_STAGE = JSON.stringify(stage)
     envObject.CYPRESS_SUPPORT = JSON.stringify(process.env.CYPRESS_SUPPORT)
 
+    if (stage === `develop`) {
+      envObject.SOCKET_IO_DEFAULT_TRANSPORT = JSON.stringify(
+        process.env.SOCKET_IO_DEFAULT_TRANSPORT || `websocket`
```

Comment:
prefix env var name with `GATSBY_` ? (to match other env vars like `GATSBY_WEBPACK_PUBLICPATH` for example)